src/clients/HubPoolClient.ts

```diff
@@ -902,18 +905,33 @@ export class HubPoolClient extends BaseAbstractClient {
     if (eventsToQuery.includes("CrossChainContractsSet")) {
       for (const event of events["CrossChainContractsSet"]) {
         const args = spreadEventWithBlockNumber(event) as CrossChainContractsSet;
-        assign(
-          this.crossChainContracts,
-          [args.l2ChainId],
-          [
-            {
-              spokePool: args.spokePool,
-              blockNumber: args.blockNumber,
-              transactionIndex: args.transactionIndex,
-              logIndex: args.logIndex,
-            },
-          ]
-        );
+        const dataToAdd = {
+          spokePool: args.spokePool,
+          blockNumber: args.blockNumber,
+          transactionIndex: args.transactionIndex,
+          logIndex: args.logIndex,
+        };
+        // If the chain is SVM then our `args.spokePool` will be set to the `solanaSpokePool.toAddressUnchecked()` in the
+        // hubpool event because our hub deals with `address` types and not byte32. Therefore, we should confirm that the
+        // `args.spokePool` is the same as the `solanaSpokePool.toAddressUnchecked()`. We can derive the `solanaSpokePool`
+        // address by using the `getDeployedAddress` function.
+        if (chainIsSvm(args.l2ChainId)) {
+          const solanaSpokePool = getDeployedAddress("SvmSpoke", args.l2ChainId);
```

Comment:
Thought bubble - do we want to reference a Solana deployment as `SvmSpoke` ? It might be a mistake to assume a 1:1 link that SVM === Solana.

There are other chains using the SVM and these could feasibly be supported by Across (i.e. [Eclipse](https://www.eclipse.xyz)), in which case we'd end up with naming inconsistencies. Should we consider naming this deployment `SolanaSpoke` instead?

(nb. I realise this is mostly an upstream question - tagging @Reinis-FRP @md0x for their thoughts on this)

Reply:
Agreed here - I was referencing the contracts repo directly - should I propose that change?